projects/libs/core-library-typescript/src/consensus/raft-consensus.ts

```diff
@@ -103,61 +112,77 @@ export class RaftConsensus {
     //
     // Note: we might consider adding transactions as the result to the "appended" event, which will require further
     // synchronization, but will make everything a wee bit faster.
-    this.node.on("committed", async (data: any) => {
-      const msgData = data.data;
 
-      const txData = msgData.msg;
+    this.node.on("committed", async (data: any) => this.onCommitted(data));
 
-      // Since we're currently storing single transactions per-block, we'd increase the block numbers for every
-      // committed entry.
-      if (this.lastBlockId == -1) {
-        const { blockId } = await this.blockStorage.getLastBlockId({});
-        this.lastBlockId = blockId;
-      }
+    this.node.on("leaderElected", () => this.onLeaderElected());
 
-      this.lastBlockId++;
-
-      await this.blockStorage.addBlock({
-        block: {
-          header: {
-            version: 0,
-            id: this.lastBlockId,
-            prevBlockId: this.lastBlockId - 1
-          },
-          tx: txData.tx,
-          modifiedAddressesJson: txData.modifiedAddressesJson,
-        }
-      });
+    this.pollForPendingTransactions();
+  }
+
+  private async onCommitted(data: any) {
+    const msg: types.ConsensusMessage = data.data;
+
+    // Since we're currently storing single transactions per-block, we'd increase the block numbers for every
+    // committed entry.
+    const block: types.Block = msg.block;
+    logger.debug("new block to be committed ${JSON.stringify(block)}");
+    await this.blockStorage.addBlock({
+      block: block
     });
+    this.lastBlockId = block.header.id;
+
+    await this.transactionPool.clearPendingTransactions({ transactions: block.transactions });
+
+    if (this.node.isLeader()) {
+      this.readyForBlockAppend = true;
+    }
+  }
 
-    this.node.on("leaderElected", () => {
-      if (this.node.isLeader()) {
-        logger.info(`Node ${this.node.id} was elected as a new leader!`);
+  private async onLeaderElected() {
+    if (this.node.isLeader()) {
+      this.readyForBlockAppend = true;
+      logger.info(`Node ${this.node.id} was elected as a new leader!`);
+    } else {
+      this.readyForBlockAppend = false;
+    }
+  }
+
+  private pollForPendingTransactions() {
+    setInterval(async () => {
+      try {
+        if (this.readyForBlockAppend) {
+          logger.debug(`attempting append of new block`);
```

Comment:
Please log the block id or hash so we can identify faulty blocks later.

Reply:
There's no specific block to log at this stage